### PR TITLE
Feature/addsUnitTests

### PR DESCRIPTION
**Pull request by cordhernandez**
adds Unit Tests for all files
#103

### CODE REVIEW

File: BlackNectar/BlackNectar/StoresTableViewController.swift [hunk 1]
```diff
@@ -103,7 +103,7 @@ class StoresTableViewController: UITableViewController, SideMenuFilterDelegate,
         
     }
     
-    fileprivate func loadStores(at coordinate: CLLocationCoordinate2D) {
+    internal func loadStores(at coordinate: CLLocationCoordinate2D) {
```
Comment (SirWellington) [inline BlackNectar/BlackNectar/StoresTableViewController.swift]: `internal` is sort of implied, so you don't really need it.

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+//
+
+@testable import BlackNectar
+
+import CoreLocation
+import XCTest
+
+class DistanceCalculationTests: XCTestCase {
+    
+    func testGetDistance() {
```
Comment (SirWellington) [inline BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:16]: Nice and clean!

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    func testMilesToMeters() {
+        
+        let miles: Double = 100.00
+        
+        XCTAssertNotNil(DistanceCalculation.milesToMeters(miles: miles))
+        
+        
+    }
+    
+    func testMetersToMiles() {
```
Comment (SirWellington) [inline BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:39]: Don't forget to test it with bad arguments. What happens when you pass in a negative number?

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        
+        let usersLatitude: CLLocationDegrees = 40.6782
+        let usersLongitude: CLLocationDegrees = -73.9442
+        let userslocation = CLLocationCoordinate2D(latitude: usersLatitude, longitude: usersLongitude)
+        
+        let restaurantsLatitude: CLLocationDegrees = 34.14341
+        let restaurantsLongitude: CLLocationDegrees = -118.392
+        let restaurantsLocation = CLLocationCoordinate2D(latitude: restaurantsLatitude, longitude: restaurantsLongitude)
+        
+        XCTAssertNotNil(DistanceCalculation.getDistance(userLocation: userslocation, storeLocation: restaurantsLocation))
```
Comment (SirWellington) [inline BlackNectar/BlackNectarTests/DistanceCalculationTests.swift]: The check should be more than "not nil". You should expected the distance to be, let's say 24 miles or whatever it actually is. This is what makes sure the distance formula works as expected.

File: BlackNectar/BlackNectarTests/SearchStoresTest.swift [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    
+    func testSearchForStoresByName() {
+        
+        let storeName: String = "Vons"
+        
+        let promise = expectation(description: "Callback will be called")
+        
+        let testCallback: ([StoresInfo]) -> Void = { stores in
+            
+            if stores.isEmpty {
```
Comment (SirWellington) [inline BlackNectar/BlackNectarTests/SearchStoresTest.swift]: Think you meant `if !stores.isEmpty {`. The promise is fulfilled if the call returns stores.

**Review by SirWellington — CHANGES_REQUESTED** [review]
Love it! Just a few touches and it's good to go.

File: BlackNectar/BlackNectar/StoresTableViewController.swift (hunk 1 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectar/StoresTableViewController.swift]: got it done!

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift (hunk 2 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:16]: thank you sir.

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:39]: + will test with negative number
+ test passes with negative number

File: BlackNectar/BlackNectarTests/SearchStoresTest.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/SearchStoresTest.swift]: + test fails with if !stores.isEmpty {
+ test passes with  if stores.isEmpty {

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift (hunk 4 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/DistanceCalculationTests.swift]: + this is taking up more time than I thought 
+ this shouldn't take me this long but for some reason it is 
+ working on it

File: BlackNectar/BlackNectarTests/SearchStoresTest.swift (hunk 5 above)
Comment (SirWellington) [reply BlackNectar/BlackNectarTests/SearchStoresTest.swift]: Hmm, perhaps the test is trying to load stores somewhere in the world where there are none. Have you tried setting a breakpoint to see whether stores actually come back or not?

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift (hunk 3 above)
Comment (SirWellington) [reply BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:39]: nice. The point was to add a separate test case specifically for those "Bad Arguments"
`testMetersToMilesWithBadArgs()`

File: BlackNectar/BlackNectarTests/SearchStoresTest.swift (hunk 5 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/SearchStoresTest.swift]: Fixed this issue

File: BlackNectar/BlackNectarTests/DistanceCalculationTests.swift (hunk 3 above)
Comment (cordhernandez) [reply BlackNectar/BlackNectarTests/DistanceCalculationTests.swift:39]: Oh ok
will add separate test case

**Pull request merged**